### ISSUE TITLE
Az.StorageSync: New-AzStorageSyncServerEndpoint is not picking up -TierFilesOlderThanDays parameter

### ISSUE THREAD

**Issue by murakest** [Azure/azure-powershell#9551]
## Description
New-AzStorageSyncServerEndpoint powershell cmdlet is not picking up -TierFilesOlderThanDays parameter. Command completes but value for this parameter is null.

Set-AzStorageSyncServerEndpoint cmdlet is picking this parameter without any issues.
Az module 2.3.2
Az.StorageSync module 1.1.0
Region: west europe

## Steps to reproduce
```
New-AzStorageSyncServerEndpoint -Name $registeredServer.FriendlyName -SyncGroup $syncGroup -ServerResourceId $registeredServer.ResourceId -ServerLocalPath "$serverEndpointPath" -CloudTiering -VolumeFreeSpacePercent $volumeFreeSpacePercentage -TierFilesOlderThanDays 90

Output:
CloudTiering                                : On
VolumeFreeSpacePercent                      : 20
TierFilesOlderThanDays                      :
OfflineDataTransfer                         : Off
OfflineDataTransferShareName                :
OfflineDataTransferStorageAccountResourceId :
OfflineDataTransferStorageAccountTenantId   :

set-AzStorageSyncServerEndpoint -ResourceGroupName $rgname -StorageSyncServiceName $storagesyncname -SyncGroupName $syncgroup -Name $srvendpointname -CloudTiering -VolumeFreeSpacePercent 20 -TierFilesOlderThanDays 90

Output:
CloudTiering                                : On
VolumeFreeSpacePercent                      : 20
TierFilesOlderThanDays                      : 90
OfflineDataTransfer                         : Off
OfflineDataTransferShareName                :
OfflineDataTransferStorageAccountResourceId :
OfflineDataTransferStorageAccountTenantId   :
```

## Environment data
```
Name                           Value
----                           -----
PSVersion                      5.1.17134.765
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.765
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

## Module versions
```
Directory: %userprofile%\Documents\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands                                   
---------- -------    ----                                ----------------                                   
Script     1.5.3      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAuto...
Script     1.0.1      Az.Aks                              {Get-AzAks, New-AzAks, Remove-AzAks, Import-AzAk...
Script     1.1.0      Az.AnalysisServices                 {Resume-AzAnalysisServicesServer, Suspend-AzAnal...
Script     1.1.0      Az.ApiManagement                    {Add-AzApiManagementApiToProduct, Add-AzApiManag...
Script     1.0.0      Az.ApplicationInsights              {Get-AzApplicationInsights, New-AzApplicationIns...
Script     1.2.2      Az.Automation                       {Get-AzAutomationHybridWorkerGroup, Remove-AzAut...
Script     1.1.0      Az.Batch                            {Remove-AzBatchAccount, Get-AzBatchAccount, Get-...
Script     1.0.0      Az.Billing                          {Get-AzBillingInvoice, Get-AzBillingPeriod, Get-...
Script     1.3.0      Az.Cdn                              {Get-AzCdnProfile, Get-AzCdnProfileSsoUrl, New-A...
Script     1.1.1      Az.CognitiveServices                {Get-AzCognitiveServicesAccount, Get-AzCognitive...
Script     2.3.0      Az.Compute                          {Remove-AzAvailabilitySet, Get-AzAvailabilitySet...
Script     2.2.0      Az.Compute                          {Remove-AzAvailabilitySet, Get-AzAvailabilitySet...
Script     1.0.1      Az.ContainerInstance                {New-AzContainerGroup, Get-AzContainerGroup, Rem...
Script     1.0.1      Az.ContainerRegistry                {New-AzContainerRegistry, Get-AzContainerRegistr...
Script     1.1.1      Az.DataFactory                      {Set-AzDataFactoryV2, Update-AzDataFactoryV2, Ge...
Script     1.0.0      Az.DataLakeAnalytics                {Get-AzDataLakeAnalyticsDataSource, New-AzDataLa...
Script     1.2.1      Az.DataLakeStore                    {Get-AzDataLakeStoreTrustedIdProvider, Remove-Az...
Script     1.0.0      Az.DeploymentManager                {Get-AzDeploymentManagerArtifactSource, New-AzDe...
Script     1.0.0      Az.DevTestLabs                      {Get-AzDtlAllowedVMSizesPolicy, Get-AzDtlAutoShu...
Script     1.1.1      Az.Dns                              {Get-AzDnsRecordSet, New-AzDnsRecordConfig, Remo...
Script     1.1.0      Az.Dns                              {Get-AzDnsRecordSet, New-AzDnsRecordConfig, Remo...
Script     1.2.0      Az.EventGrid                        {New-AzEventGridTopic, Get-AzEventGridTopic, Set...
Script     1.1.1      Az.EventGrid                        {New-AzEventGridTopic, Get-AzEventGridTopic, Set...
Script     1.2.0      Az.EventHub                         {New-AzEventHubNamespace, Get-AzEventHubNamespac...
Script     1.1.0      Az.FrontDoor                        {New-AzFrontDoor, Get-AzFrontDoor, Set-AzFrontDo...
Script     1.0.0      Az.FrontDoor                        {New-AzFrontDoor, Get-AzFrontDoor, Set-AzFrontDo...
Script     2.0.0      Az.HDInsight                        {Get-AzHDInsightJob, New-AzHDInsightSqoopJobDefi...
Script     1.1.0      Az.IotHub                           {Add-AzIotHubKey, Get-AzIotHubEventHubConsumerGr...
Script     1.2.0      Az.KeyVault                         {Add-AzKeyVaultCertificate, Update-AzKeyVaultCer...
Script     1.2.1      Az.LogicApp                         {Get-AzIntegrationAccountAgreement, Get-AzIntegr...
Script     1.1.0      Az.MachineLearning                  {Move-AzMlCommitmentAssociation, Get-AzMlCommitm...
Script     1.0.0      Az.MarketplaceOrdering              {Get-AzMarketplaceTerms, Set-AzMarketplaceTerms}   
Script     1.1.0      Az.Media                            {Sync-AzMediaServiceStorageKey, Set-AzMediaServi...
Script     1.2.1      Az.Monitor                          {Get-AzMetricDefinition, Get-AzMetric, Remove-Az...
Script     1.10.0     Az.Network                          {Add-AzApplicationGatewayAuthenticationCertifica...
Script     1.9.0      Az.Network                          {Add-AzApplicationGatewayAuthenticationCertifica...
Script     1.1.0      Az.NotificationHubs                 {Get-AzNotificationHub, Get-AzNotificationHubAut...
Script     1.3.0      Az.OperationalInsights              {New-AzOperationalInsightsAzureActivityLogDataSo...
Script     1.2.0      Az.OperationalInsights              {New-AzOperationalInsightsAzureActivityLogDataSo...
Script     1.1.1      Az.PolicyInsights                   {Get-AzPolicyEvent, Get-AzPolicyState, Get-AzPol...
Script     1.1.0      Az.PowerBIEmbedded                  {Remove-AzPowerBIWorkspaceCollection, Get-AzPowe...
Script     1.4.1      Az.RecoveryServices                 {Get-AzRecoveryServicesBackupProperty, Get-AzRec...
Script     1.1.0      Az.RedisCache                       {Remove-AzRedisCachePatchSchedule, New-AzRedisCa...
Script     1.0.1      Az.Relay                            {New-AzRelayNamespace, Get-AzRelayNamespace, Set...
Script     1.5.0      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignmen...
Script     1.4.0      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignmen...
Script     1.2.0      Az.ServiceBus                       {New-AzServiceBusNamespace, Get-AzServiceBusName...
Script     1.1.1      Az.ServiceFabric                    {Add-AzServiceFabricApplicationCertificate, Add-...
Script     1.1.0      Az.ServiceFabric                    {Add-AzServiceFabricApplicationCertificate, Add-...
Script     1.0.2      Az.SignalR                          {New-AzSignalR, Get-AzSignalR, Get-AzSignalRKey,...
Script     1.12.0     Az.Sql                              {Get-AzSqlDatabaseTransparentDataEncryption, Get...
Script     1.11.0     Az.Sql                              {Get-AzSqlDatabaseTransparentDataEncryption, Get...
Script     1.4.0      Az.Storage                          {Get-AzStorageAccount, Get-AzStorageAccountKey, ...
Script     1.3.0      Az.Storage                          {Get-AzStorageAccount, Get-AzStorageAccountKey, ...
Script     1.1.0      Az.StorageSync                      {Invoke-AzStorageSyncCompatibilityCheck, New-AzS...
Script     0.8.1      Az.StorageSync                      {Invoke-AzStorageSyncCompatibilityCheck, New-AzS...
Script     1.0.0      Az.StreamAnalytics                  {Get-AzStreamAnalyticsFunction, Get-AzStreamAnal...
Script     1.0.1      Az.TrafficManager                   {Add-AzTrafficManagerCustomHeaderToEndpoint, Rem...
Script     1.3.0      Az.Websites                         {Get-AzAppServicePlan, Set-AzAppServicePlan, New...
Script     1.2.2      Az.Websites                         {Get-AzAppServicePlan, Set-AzAppServicePlan, New...
```

**Comment by markcowl** [comment]
@ankushbindlish2 Can you take a look?

**Comment by ankushbindlish2** [comment]
@murakest , would you remove old StorageSync 0.8.1 module and try the command again.

**Comment by ankushbindlish2** [comment]
We have fixed this issue as part of this pull request https://github.com/Azure/azure-powershell/pull/9284

**Comment by murakest** [comment]
@ankushbindlish2, I have tested again and still faced the same issue. I have deployed vanilla server, installed AZ 2.4.0 module that has az.storage sync 1.1.0. Everything on azure is already pre-created and I only execute code to create file server endpoints and once I check in azure portal the "cache files option for files older then" is not set up.

New-AzStorageSyncServerEndpoint -Name $registeredServer.FriendlyName -SyncGroup $syncGroup -ServerResourceId $registeredServer.ResourceId -ServerLocalPath "$serverEndpointPath" -CloudTiering -VolumeFreeSpacePercent $volumeFreeSpacePercentage -TierFilesOlderThanDays 90
